tests/integration/test_msgraph/test_API/test_API.py

```diff
@@ -0,0 +1,530 @@
+local_internal_options = {MODULESD_DEBUG: '2'}
+
+
+@pytest.fixture(scope="session")
+def proxy_setup():
+    mocks_path = Path(CONFIGS_PATH, 'responses.json')
+    m365proxy = subprocess.Popen(["/tmp/m365proxy/m365proxy", "--mocks-file", mocks_path])
+    # Configurate proxy for Wazuh (will only work for systemctl start/restart)
+    subprocess.run("systemctl set-environment http_proxy=http://localhost:8000", shell=True)
+    remove_file(os.path.join(WAZUH_PATH, 'var', 'wodles', 'ms-graph-tenant_id-resource_name-resource_relationship'))
+
+    yield
+
+    subprocess.run("systemctl unset-environment http_proxy", shell=True)
+    m365proxy.kill()
+    m365proxy.wait()
+
+# Tests
+@pytest.mark.parametrize('test_configuration, test_metadata', zip(t1_configurations, t1_configuration_metadata), ids=t1_case_ids)
+def test_future_events_yes(test_configuration, test_metadata, set_wazuh_configuration, configure_local_internal_options,
+                 truncate_monitored_files, daemons_handler, wait_for_msgraph_start, proxy_setup):
+    '''
+    description: Check 'ms-graph' behavior when `only_future_events` tag is set to yes.
+    wazuh_min_version: 4.6.0
+
+    tier: 0
+
+    parameters:
+        - test_configuration:
+            type: data
+            brief: Configuration used in the test.
+        - test_metadata:
+            type: data
+            brief: Configuration cases.
+        - set_wazuh_configuration:
+            type: fixture
+            brief: Configure a custom environment for testing.
+        - configure_local_internal_options:
+            type: fixture
+            brief: Set internal configuration for testing.
+        - truncate_monitored_files:
+            type: fixture
+            brief: Reset the 'ossec.log' file and start a new monitor.
+        - daemons_handler:
+            type: fixture
+            brief: Manages daemons to reset Wazuh.
+        - wait_for_msgraph_start:
+            type: fixture
+            brief: Checks integration start message does not appear.
+        - proxy_setup:
+            type: fixture
+            brief: Setups the API proxy application.
+
+    assertions:
+        - Verify that when the `only_future_events` option is set to `yes`, the ms-graph module saves a bookmark, 
+        and after a restart, it waits for a first scan.
+
+    input_description: A configuration template is contained in an external YAML file
+                       (config_API.yaml). That template is combined with different test cases defined in
+                       the module. Those include configuration settings for the 'ms-graph' module.
+
+    expected_output:
+        - r'.*wazuh-modulesd:ms-graph.*Bookmark updated'
+        - r'.*wazuh-modulesd:ms-graph.*seconds to run first scan'
+    '''
+
+    wazuh_log_monitor = FileMonitor(WAZUH_LOG_PATH)
+
+    wazuh_log_monitor.start(callback=callbacks.generate_callback(r".*wazuh-modulesd:ms-graph.*Bookmark updated"))
```

Comment:
We are placing all patterns in the framework. Don't change this now, we will fix this later as part of another issue.